engine/common/imagelib/img_ktx2.c

```diff
@@ -0,0 +1,188 @@
+/*
+img_ktx2.c - KTX2 format load
+Copyright (C) 2023 Provod
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+*/
+
+#include "imagelib.h"
+#include "xash3d_mathlib.h"
+#include "ktx2.h"
+
+static pixformat_t Image_KTX2Format( uint32_t ktx2_format ) {
+	switch (ktx2_format) {
+		case KTX2_FORMAT_BC4_UNORM_BLOCK: return PF_BC4_UNSIGNED;
+		case KTX2_FORMAT_BC4_SNORM_BLOCK: return PF_BC4_SIGNED;
+		case KTX2_FORMAT_BC5_UNORM_BLOCK: return PF_BC5_UNSIGNED;
+		case KTX2_FORMAT_BC5_SNORM_BLOCK: return PF_BC5_SIGNED;
+		case KTX2_FORMAT_BC6H_UFLOAT_BLOCK: return PF_BC6H_UNSIGNED;
+		case KTX2_FORMAT_BC6H_SFLOAT_BLOCK: return PF_BC6H_SIGNED;
+		case KTX2_FORMAT_BC7_UNORM_BLOCK: return PF_BC7_UNORM;
+		case KTX2_FORMAT_BC7_SRGB_BLOCK: return PF_BC7_SRGB;
+		default: return PF_UNKNOWN;
+	}
+}
+
+// FIXME this codebase has too many copies of this function:
+// - ref_gl has one
+// - ref_vk has one
+// - ref_soft has one
+// - img_dds has one
+// - many more places probably have one too
+// - and now img_ktx2 also has one!
```

Comment:
Haha. For this, I think it can be moved to img_utils.c.

We need some ref_common library. I had plans on making one, since ref_soft and ref_gl could share some code. Absolutely optional, as I like how current RefAPI gives so much freedom, though it's a bit bloated.

Reply:
I can move it to `img_utils.c` to at least deduplicate this code for imagelib.

It still won't be accessible from the renderers. That part I feel is out of scope for me for now at least, unless there's a simple solution. Eventually, `ref_common` would be nice for sure. Or `ref_stdlib`, or whatever.